MBBSEmu.Tests/CPU/SHR_Tests.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Iced.Intel;
+using MBBSEmu.CPU;
+using MBBSEmu.Extensions;
+using Xunit;
+using static Iced.Intel.AssemblerRegisters;
+
+namespace MBBSEmu.Tests.CPU
+{
+    public class SHR_Tests : CpuTestBase
```

Comment:
We should add tests for shifts of up to the max value per type to ensure it keeps the correct byte with the type extension code. 

Also, this logic:

```
To reduce the maximum execution time, the 386 DX microprocessor does not allow shift counts greater 
than 31. If a shift count greater than 31 is attempted, only the bottom five bits of the shift count are 
used. (The 8086 uses all eight bits of the shift count.) 
```